src/services/Store/store/vault.slice.test.ts

```diff
@@ -0,0 +1,88 @@
+const decryptedData =
+  '{"version":"v1.0.0","mtime":1608048741382,"accounts":{},"addressBook":{"a1acf1f2-0380-5bd6-90c3-2b4a0974a6fe":{"label":"MyCrypto Tip Jar","address":"0x4bbeEB066eD09B7AEd07bF39EEe0460DFa261520","notes":"Toss us a coin!","network":"Ethereum","uuid":"a1acf1f2-0380-5bd6-90c3-2b4a0974a6fe"}},"assets":{},"contracts":{},"networks":{},"notifications":{},"settings":{"fiatCurrency":"USD","darkMode":false,"dashboardAccounts":[],"excludedAssets":[],"inactivityTimer":1800000,"rates":{},"language":"en"},"password":"b9c950640e1b3740e98acb93e669c65766f6670dd1609ba91ff41052ba48c6f3","networkNodes":{},"userActions":{}}';
+
+Date.now = jest.fn(() => 1608048741382);
+
+describe('encryptionWorker()', () => {
+  it('encrypts existing state and resets app state', () => {
+    return expectSaga(encryptionWorker, encrypt(hashedPassword))
+      .withState({ legacy: marshallState(JSON.parse(decryptedData)) })
+      .call(encryptData, decryptedData, hashedPassword)
+      .put(setEncryptedData(encryptedData))
+      .silentRun();
+  });
+});
+
+describe('decryptionWorker()', () => {
+  it('decrypts existing state and imports', () => {
+    return expectSaga(decryptionWorker, decrypt(hashedPassword))
+      .withState({
+        legacy: { networks: fNetworks, assets: fAssets, accounts: fAccounts },
+        vault: {
+          data: encryptedData
+        }
+      })
+      .call(decryptData, encryptedData, hashedPassword)
+      .put(clearEncryptedData())
+      .put(importState(decryptedData))
+      .silentRun();
+  });
+
+  it('goes into error state if password is wrong', () => {
+    return expectSaga(decryptionWorker, decrypt(wrongPassword))
+      .withState({
+        legacy: { networks: fNetworks, assets: fAssets, accounts: fAccounts },
+        vault: {
+          data: encryptedData
+        }
+      })
```

Comment:
use / introduce mockAppState from test-utils

Reply:
Getting massive circular dependency issues when trying to introduce the `mockAppState`. Would prefer fixing when one of the PRs are merged.